src/imitation/data/buffer.py

```diff
@@ -273,14 +273,36 @@ def __init__(
         """
         params = (obs_shape, act_shape, obs_dtype, act_dtype)
         if venv is not None:
-            if not all(x is None for x in params):
-                raise ValueError(
-                    "Cannot specify both shape/dtype and also environment.",
-                )
-            obs_shape = tuple(venv.observation_space.shape)
-            act_shape = tuple(venv.action_space.shape)
-            obs_dtype = venv.observation_space.dtype
-            act_dtype = venv.action_space.dtype
+            if venv.observation_space.shape is not None:
+                if obs_shape is not None:
+                    raise ValueError(
+                        "Cannot specify both observation shape and also environment "
+                        "with an observation space that has a shape.",
+                    )
+                obs_shape = tuple(venv.observation_space.shape)
```

Comment:
Minor preference for consistently adding whitespace between these `if` statements. IE:

```python
    obs_shape = tuple(venv.observation_space.shape)

if venv.observation_space.dtype is not None:
```

Reply:
You are right, it reads a lot easier this way!